core/domain/stats_services.py

```diff
@@ -1024,3 +1026,237 @@ def _get_calc_output(exploration_id, state_name, calculation_id):
+            learner_answer_details_model.accumulated_answer_info_json_size_bytes = ( #pylint: disable=line-too-long
+                learner_answer_details.accumulated_answer_info_json_size_bytes)
+            learner_answer_details_model.put()
+        else:
+            learner_answer_details_model.delete()
+            stats_models.LearnerAnswerDetailsModel.create_model_instance(
+                learner_answer_details.state_reference,
+                learner_answer_details.entity_type,
+                learner_answer_details.interaction_id,
+                [learner_answer_info.to_dict() for learner_answer_info
+                 in learner_answer_details.learner_answer_info_list],
+                learner_answer_details.learner_answer_info_schema_version,
+                learner_answer_details.accumulated_answer_info_json_size_bytes)
+    else:
+        stats_models.LearnerAnswerDetailsModel.create_model_instance(
+            learner_answer_details.state_reference,
+            learner_answer_details.entity_type,
+            learner_answer_details.interaction_id,
+            [learner_answer_info.to_dict() for learner_answer_info
+             in learner_answer_details.learner_answer_info_list],
+            learner_answer_details.learner_answer_info_schema_version,
+            learner_answer_details.accumulated_answer_info_json_size_bytes)
+
+
+def record_learner_answer_info(
+        state_reference, entity_type, interaction_id, answer, answer_details):
+    """Records the new learner answer info received from the learner in the
+    model and then saves it.
+
+    Args:
+        state_reference: str. This is used to refer to a state
+            in an exploration or question. For an exploration the
+            value will be equal to 'exp_id.state_name' and for question
+            this will be equal to 'question_id'.
+        entity_type: str. The type of entity i.e ENTITY_TYPE_EXPLORATION
+            or ENTITY_TYPE_QUESTION, which are declared in feconf.py.
+        interaction_id: str. The ID of the interaction.
+        answer: dict or list or str or int or bool. The answer which is
+            submitted by the learner. Actually type of the answer is
+            interaction dependent, like TextInput interactions have
+            string type answer, NumericInput have int type answers etc.
+        answer_details: str. The details the learner will submit when the
+            learner will be asked questions like 'Hey how did you land on
+            this answer', 'Why did you pick that answer' etc.
+    """
+    learner_answer_details = get_learner_answer_details(
+        state_reference, entity_type)
+    if not learner_answer_details:
+        learner_answer_details = stats_domain.LearnerAnswerDetails(
+            state_reference, entity_type, interaction_id, [], 0)
+    learner_answer_info_id = (
+        stats_domain.LearnerAnswerInfo.get_learner_answer_info_id())
```

Comment:
Can we rename the method to get_new_...? Otherwise, unclear that it's creating a new thing.

Reply:
Done